<?xml version="1.0"?>
<!-- Test the projectile estimator by running back real Vicon data from a bag file and estimating the state. The estimates are recorded to a new bag file. -->

<launch>
  <arg name="bagfile"/>

  <!-- use simulation time -->
  <param name="/use_sim_time" value="true" type="bool"/>

  <!-- Estimation -->
  <param name="robot_proc_var" type="double" value="1000"/>
  <param name="robot_meas_var" type="double" value="0.001"/>

  <!-- MRT -->
  <node pkg="upright_ros_interface" type="mrt_node" name="mrt_node" output="screen" required="True" args="$(arg config)" />

  <!-- Record the estimation results -->
  <node pkg="rosbag" type="record" name="recorder" output="screen" required="true" args="--all -o $(find upright_ros_interface)/robot_estimation"/>

  <!-- Play back the robot feedback and the MPC feedback -->
  <!-- We want to do estimation based on the existing feedback and inputs -->
  <node pkg="rosbag" type="play" name="player" required="true" output="screen" args="$(arg bagfile) --loop --clock --topics /ridgeback/joint_states /ur10/joint_states /mobile_manipulator_mpc_policy" />
</launch>
